videoanalyser/app/src/main/java/com/example/videoanalyser/MainActivity.java

```diff
@@ -0,0 +1,132 @@
+package com.example.videoanalyser;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.mlkit.vision.common.InputImage;
+import com.google.mlkit.vision.text.Text;
+import com.google.mlkit.vision.text.TextRecognition;
+import com.google.mlkit.vision.text.TextRecognizer;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+import static android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_FRAME_COUNT;
+
+public class MainActivity extends AppCompatActivity {
+    private TextView filePath;
+    private Button filePickerBtn;
+    private Button analyzeBtn;
+    private Intent fileIntent;
+    private MediaMetadataRetriever mediaMetadataRetriever;
+    private Uri fileUri;
+    private  Integer totalFrames=0;
+    private  Integer safeFrames=10;
+    private String rotationOfVideo;
+    private InputImage imageHolder;
+    private TextRecognizer recognizer;
+    private ArrayList<String> results = new ArrayList<String>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        filePath = findViewById(R.id.selectedFilePath);
+        filePickerBtn = findViewById(R.id.fileSelectButton);
+        analyzeBtn = findViewById(R.id.analyze);
+        recognizer = TextRecognition.getClient();
+
+        filePickerBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                fileIntent = new Intent(Intent.ACTION_GET_CONTENT);
+                fileIntent.setType("*/*");
+                startActivityForResult(fileIntent, 10);
+            }
+
+        });
+
+        analyzeBtn.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.P)
```

Comment:
Nit: Just curious, where does this come from/why do we need this annotation?

Reply:
It was compiler suggested. Tells programmer that an android Pie(9) API is needed.
I think that would limit the number of divices, I will look into this more.

Reply:
The function `getFramesAtIndex` at Line 86 reuires this API level.
`List<Bitmap> frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames-safeFrames))